File: eid-mobile-android/data/src/main/java/com/digitall/eid/data/models/network/applications/all/ApplicationDetailsResponse.kt
/**
 * Please follow code style when editing project
 * Please follow principles of clean architecture
 * Created 2024 by Roman Kryvolapov
 **/
package com.digitall.eid.data.models.network.applications.all

import com.google.gson.annotations.SerializedName

data class ApplicationDetailsResponse(
    @SerializedName("id") val id: String?,
    @SerializedName("xml") val xml: String?,
    @SerializedName("email") val email: String?,
    @SerializedName("status") val status: String?,
    @SerializedName("reasonId") val reasonId: String?,
    @SerializedName("createDate") val createDate: String?,
    @SerializedName("lastName") val lastName: String?,
    @SerializedName("firstName") val firstName: String?,
    @SerializedName("secondName") val secondName: String?,
    @SerializedName("deviceId") val deviceId: String?,
    @SerializedName("reasonText") val reasonText: String?,
    @SerializedName("phoneNumber") val phoneNumber: String?,
    @SerializedName("eidentityId") val eidentityId: String?,
    @SerializedName("serialNumber") val serialNumber: String?,
    @SerializedName("certificateId") val certificateId: String?,
    @SerializedName("applicationNumber") val applicationNumber: String?,
    @SerializedName("submissionType") val submissionType: String,
    @SerializedName("applicationType") val applicationType: String?,
    @SerializedName("eidAdministratorName") val eidAdministratorName: String?,
    @SerializedName("eidAdministratorOfficeName") val eidAdministratorOfficeName: String?,
    @SerializedName("identityType") val identityType: String?,
    @SerializedName("identityNumber") val identityNumber: String?,
    @SerializedName("identityIssueDate") val identityIssueDate: String?,
    @SerializedName("identityValidityToDate") val identityValidityToDate: String?,
    @SerializedName("paymentAccessCode") val paymentAccessCode: String?,
)

File: eid-mobile-ios/eID-Mobile-iOS/02_Data/Entities/ElectronicIdentity/Applications/ApplicationDetailsResponse.swift
//
//  ApplicationDetailsResponse.swift
//  eID-Mobile-iOS
//
//  Created by Sunay Gyultekin on 11.03.24.
//

import Foundation


struct ApplicationDetailsResponse: Codable {
    // MARK: - Properties
    var id: String
    var firstName: String
    var secondName: String?
    var lastName: String
    var applicationType: EIDApplicationType
    var certificateId: String?
    var serialNumber: String?
    var status: ApplicationStatus?
    var createDate: String
    var deviceId: String
    var eidAdministratorName: String
    var eidAdministratorOfficeName: String
    var eidentityId: String?
    var email: String?
    var phoneNumber: String?
    var reasonId: String?
    var reasonText: String?
    var xml: String
    var submissionType: ApplicationSubmissionType
    var identityNumber: String?
    var identityType: String?
    var identityIssueDate: String?
    var identityValidityToDate: String?
    var applicationNumber: String?
    var paymentAccessCode: String?
}


enum ApplicationSubmissionType: String, Codable, CaseIterable {
    case desk = "DESK"
    case baseProfile = "BASE_PROFILE"
    case eid = "EID"
    case persoCentre = "PERSO_CENTRE"
}
